### PR TITLE
Enable agent-based instrumentation by default

### LINKED ISSUE #23297: Enable agent-based instrumentation by default
Currently, the agent-based instrumentation is under a feature flag. There are some rough edges that prevent us from enabling it:

- [x] Foreground daemons are not supported
- [x] Embedded test runner doesn't have agent applied
- [x] ~~Running Testkit tests in embedded mode (`withDebug(true)`) is not supported~~ (Tracked in a separate issue #23817)
- [x] Some expectations for the TAPI tests are off
- [x] Test coverage should be improved

### PR DESCRIPTION
**Pull request by mlopatkin**
Fixes #23297

### CODE REVIEW

**Comment by mlopatkin** [comment]
@bot-gradle test RfN

File: build-logic/integration-testing/src/main/kotlin/gradlebuild/integrationtests/shared-configuration.kt [hunk 1]
```diff
@@ -179,16 +179,16 @@ fun Project.createTestTask(name: String, executer: String, sourceSet: SourceSet,
 
 private
 fun IntegrationTest.setUpAgentIfNeeded(testType: TestType, executer: String) {
+    if (executer == "embedded") {
```
Comment (bamboo) [inline build-logic/integration-testing/src/main/kotlin/gradlebuild/integrationtests/shared-configuration.kt:182]: 🎨 I think I could have used a comment here saying something like _Add the java agents to the test JVM when running integration tests using the embedded Gradle executer._

**Review by bamboo — APPROVED** [review]

**Comment by mlopatkin** [comment]
@bot-gradle test APTN

**Comment by mlopatkin** [comment]
@bot-gradle test APT

**Comment by mlopatkin** [comment]
@bot-gradle test and merge

**Pull request merged**